src/Native/Runtime/RuntimeInstance.cpp

```diff
@@ -821,40 +821,34 @@ enum RuntimeHelperKind
 // The dictionary codegen expects a pointer that points at a memory location that points to the method pointer
 // Create indirections for all helpers used below
 
-#define DECLARE_INDIRECTION(HELPER_NAME) \
-    EXTERN_C void HELPER_NAME(); \
-    const PTR_VOID indirection_##HELPER_NAME = (PTR_VOID)&HELPER_NAME
-#define DECLARE_INDIRECTIONvpp(HELPER_NAME) \
-    EXTERN_C void HELPER_NAME(void*, void*); \
-    const PTR_VOID indirection_##HELPER_NAME = (PTR_VOID)&HELPER_NAME
-#define DECLARE_INDIRECTIONppp(HELPER_NAME) \
-    EXTERN_C void* HELPER_NAME(void*, void*); \
+#define DECLARE_INDIRECTION(RET_TYPE, HELPER_NAME, ARGS) \
+    EXTERN_C RET_TYPE HELPER_NAME ARGS; \
     const PTR_VOID indirection_##HELPER_NAME = (PTR_VOID)&HELPER_NAME
 
 #define INDIRECTION(HELPER_NAME) ((PTR_VOID)&indirection_##HELPER_NAME)
 
-DECLARE_INDIRECTION(RhpNewFast);
-DECLARE_INDIRECTION(RhpNewFinalizable);
+DECLARE_INDIRECTION(void, RhpNewFast, ());
```

Comment:
These signatures are a bit off. I assume it doesn't matter because the symbol is defined in C++?

Maybe add a `#define DECLARE_INDIRECTION(HELPER_NAME) DECLARE_INDIRECTION(void, HELPER_NAME, ())` and use that for the "don't care" symbols?

Reply:
> I'm kind of surprised we only miss the guard symbol and not `__stack_chk_fail`.
It may well do but wasm-ld bails after a few errors.  The other symbols are undefined also according to `llvm-nm`

```
         U __stack_chk_fail
         U __stack_chk_guard
         U __stack_pointer
```

Reply:
> These signatures are a bit off. I assume it doesn't matter because the symbol is defined in C++?

I dont know, doesn't seem like any of the indirection symbols are used outside of `libPortableRuntime.a`.  They are just warnings and we don't necessarily need this change.  Without it `wasm-ld`, the llvm wasm linker, produces messages such as 
```
wasm-ld : warning : function signature mismatch: RhTypeCast_IsInstanceOfClass [E:\GitHub\corert\tests\src\Simple\HelloWasm\HelloWasm.csproj]
  >>> defined as () -> void in E:\GitHub\corert\tests\src\Simple\HelloWasm\obj\Debug\wasm\native\HelloWasm.bc
  >>> defined as (i32, i32) -> i32 in lto.tmp
```
`lto` I think is an abbreviation for link time optimisation.  But its complaining about the base symbol, not the `indirection_*` symbol which I don't get.  For review I've checked in a change as suggested (except I don't think macro overloading is supported so used a different name).